demos/common/cpp/models/src/detection_model_retinaface_pt.cpp

```diff
@@ -0,0 +1,250 @@
+        }
+        else {
+            continue;
+        }
+    }
+
+    if (outputsNames[OT_BBOX] == "" || outputsNames[OT_SCORES] == "") {
+        throw std::runtime_error("Bbox or cls layers are not found");
+    }
+
+    if (outputsNames.size() != 2 && outputsNames.size() != 3) {
+        throw std::logic_error("Expected 2 or 3 output blobs");
+    }
+
+    priors = generatePriorData();
+}
+
+
+std::vector<uint32_t> ModelRetinaFacePT::doThresholding(const InferenceEngine::MemoryBlob::Ptr& rawData, const float confidenceThreshold) {
+    std::vector<uint32_t> indicies;
+    auto desc = rawData->getTensorDesc();
+    auto sz = desc.getDims();
+    InferenceEngine::LockedMemory<const void> outputMapped = rawData->rmap();
+    const float *memPtr = outputMapped.as<float*>();
+
+    for (uint32_t x = 0; x < sz[1]; ++x) {
+        auto idx = (x * sz[2] + 1);
+        auto score = memPtr[idx];
+        if (score >= confidenceThreshold) {
+            indicies.push_back(x);
+        }
+    }
+
+    return indicies;
+}
+
+std::vector<float> ModelRetinaFacePT::getFilteredScores(const InferenceEngine::MemoryBlob::Ptr& rawData, const std::vector<uint32_t>& indicies) {
+    InferenceEngine::LockedMemory<const void> outputMapped = rawData->rmap();
+    const float *memPtr = outputMapped.as<float*>();
+    auto desc = rawData->getTensorDesc();
+    auto sz = desc.getDims();
+
+    std::vector<float> scores;
+    scores.reserve(indicies.size());
+
+    for (auto i : indicies) {
+        scores.push_back(memPtr[i*sz[2] + 1]);
+    }
+    return scores;
+}
+
+std::vector<cv::Point2f> ModelRetinaFacePT::getFilteredLandmarks(const InferenceEngine::MemoryBlob::Ptr& rawData, const std::vector<uint32_t>& indicies, int imgWidth, int imgHeight) {
+    InferenceEngine::LockedMemory<const void> outputMapped = rawData->rmap();
+    const float *memPtr = outputMapped.as<float*>();
+    auto desc = rawData->getTensorDesc();
+    auto sz = desc.getDims();
+
+    std::vector<cv::Point2f> landmarks(landmarksNum*indicies.size());
+
+    for (size_t i = 0; i < indicies.size(); i++) {
+        uint32_t idx = indicies[i];
+        auto& prior = priors[idx];
+        for (size_t j = 0; j < landmarksNum; j++) {
+            landmarks[i*landmarksNum + j].x = (prior.cX + memPtr[idx*sz[2] + j*2] * variance[0] * prior.width) * imgWidth;
+            landmarks[i*landmarksNum + j].y = (prior.cY + memPtr[idx*sz[2] + j*2 + 1] * variance[0] * prior.height) * imgHeight;
+        }
+    }
+    return landmarks;
+}
+
+std::vector<ModelRetinaFacePT::Box> ModelRetinaFacePT::generatePriorData() {
+    float globalMinSizes[][2] = { {16, 32}, {64, 128}, {256, 512} };
+    float steps[] = { 8., 16., 32. };
+    std::vector<ModelRetinaFacePT::Box> anchors;
+    for (size_t stepNum = 0; stepNum < _countof(steps); stepNum++) {
+        int featureW = (int)std::round(netInputWidth / steps[stepNum]);
+        int featureH = (int)std::round(netInputHeight / steps[stepNum]);
```

Comment:
```suggestion
        const int featureW = (int)std::round(netInputWidth / steps[stepNum]);
        const int featureH = (int)std::round(netInputHeight / steps[stepNum]);
```

Reply:
I'll merge this, but frankly... is const modifier THAT important here ?